drivers/pci/pci.c

```diff
@@ -157,6 +159,46 @@ void rtc_set_checksum(int range_start, int range_end, int cks_loc)
 	CMOS_WRITE(((sum >> 0) & 0x0ff), cks_loc+1);
 }
 
+/* SMSC LPC47M157 Super I/O functions */
+/* pin and register compatible with LPC47M192 */
+void LpcEnableConfiguration(int Enable)
+{
+	IoOutputByte(0x2e, Enable ? 0x55 : 0xAA);
+}
+
+int LpcGetSerialState()
+{
+	// Select serial device
+	IoOutputByte(0x2e, 0x07);
+	IoOutputByte(0x2f, 0x04);
+
+	// Check whether device is enabled
+	IoOutputByte(0x2e, 0x30);
+	return IoInputByte(0x2f);
+}
+
+void LpcSetSerialState(int Enable, u16 Port, u8 Irq)
+{
+	// Select serial device
+	IoOutputByte(0x2e, 0x07);
+	IoOutputByte(0x2f, 0x04);
+
+	// Enable device
+	IoOutputByte(0x2e, 0x30);
+	IoOutputByte(0x2f, Enable ? 0x01 : 0x00);
+
+	// Set Serial Base
+	IoOutputByte(0x2e, 0x61);
+	IoOutputByte(0x2f, Port & 0xff);
+
+	IoOutputByte(0x2e, 0x60);
+	IoOutputByte(0x2f, Port >> 8);
+
+	// Set Serial Interrupt
+	IoOutputByte(0x2e, 0x70);
+	IoOutputByte(0x2f, Irq);
```

Comment:
nit: Consider splitting port & IRQ setup away from enabling/disabling.

The text menu never changes port info, so if this is not reset when disabling the serial, why both passing it every time serial is toggled.

You could set up the port & IRQ during startup, and enable serial.
Then, in the menu, only toggle the "enabled" - you don't have to touch port & IRQ again.

Also would help with consistency (as `LpcGetSerialState` currently returns less than `LpcSetSerialState` sets)

Reply:
Done.